docs/specification/draft/client/elicitation.mdx

```diff
@@ -151,139 +257,236 @@ To request information from a user, servers send an `elicitation/create` request
+
+The error **MUST** include a list of elicitations that are required to complete before the original can be retried.
+
+Any elicitations returned in the error **MUST** be out-of-band mode elicitations and have an `elicitationId` property.
+Servers that want elicit data from the user via form mode **SHOULD** make a separate elicitation request for each form mode elicitation.
+
+**Error Response:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "error": {
+    "code": -32604, // ELICITATION_REQUIRED
+    "message": "This request requires more information.",
+    "data": {
+      "elicitations": [
+        {
+          "mode": "oob",
+          "elicitionId": "550e8400-e29b-41d4-a716-446655440000",
+          "url": "https://oauth.example.com/authorize?client_id=abc123&response_type=code&...",
+          "message": "Authorization is required to access your Example Co files."
+        }
+      ]
+    }
   }
 }
 ```
 
 ## Message Flow
 
+### Form Mode Flow
+
 ```mermaid
 sequenceDiagram
     participant User
     participant Client
     participant Server
 
-    Note over Server,Client: Server initiates elicitation
-    Server->>Client: elicitation/create
+    Note over Server: Server initiates elicitation
+    Server->>Client: elicitation/create (mode: form)
 
-    Note over Client,User: Human interaction
-    Client->>User: Present elicitation UI
+    Note over User,Client: Present elicitation UI
     User-->>Client: Provide requested information
 
     Note over Server,Client: Complete request
-    Client-->>Server: Return user response
+    Client->>Server: Return user response
 
     Note over Server: Continue processing with new information
 ```
 
-## Request Schema
+### Out-of-Band Mode Flow
 
-The `requestedSchema` field allows servers to define the structure of the expected response using a restricted subset of JSON Schema. To simplify implementation for clients, elicitation schemas are limited to flat objects with primitive properties only:
+```mermaid
+sequenceDiagram
+    participant UserAgent as User Agent (Browser)
+    participant User
+    participant Client
+    participant Server
 
-```json
-"requestedSchema": {
-  "type": "object",
-  "properties": {
-    "propertyName": {
-      "type": "string",
-      "title": "Display Name",
-      "description": "Description of the property"
-    },
-    "anotherProperty": {
-      "type": "number",
-      "minimum": 0,
-      "maximum": 100
-    }
-  },
-  "required": ["propertyName"]
-}
-```
+    Note over Server: Server initiates elicitation
+    Server->>Client: elicitation/create (mode: oob)
 
-### Supported Schema Types
+    Client->>User: Present consent to open URL
+    User-->>Client: Provide consent
 
-The schema is restricted to these primitive types:
+    Client->>UserAgent: Open URL
+    Client->>Server: Accept response
+    Client-->>Server: elicitation/track request (optional)
```

Comment:
There's an opportunity to combine these. I recognize that there's value in leaning into the concept of general progress tracking but at the same time, there MUST be an accept response so why not allow the tracking to optionally piggy back on that?